sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/UdfUdafRegister.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.beam.sdk.transforms.Combine;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+
+/** Register for UDF and UDAF. */
+public interface UdfUdafRegister {
+  default Map<String, Class<? extends BeamSqlUdf>> getEvalUdfs() {
+    return new HashMap<>();
+  }
+
+  default Map<String, SerializableFunction<?, ?>> getSerializeUdfs() {
+    return new HashMap<>();
```

Comment:
I think it should return an immutable empty map in this case (`Collections.emptyMap()`)